lib/engine/game/g_1822_pnw/game.rb

```diff
@@ -696,31 +696,19 @@ def upgrades_to?(from, to, special = false, selected_company: nil)
           super
         end
 
-        def payout_companies
+        def close_corporation(corporation, quiet: false)
           super
-          # Check on stock drop private
-          company = company_by_id('P16')
-          return unless company.owner.is_a?(Corporation)
-
-          payment = 10
-          if company.owner.cash >= payment
-            @log << "#{company.owner.name} spends #{format_currency(payment)} because of #{company.name}"
-            company.owner.spend(payment, bank)
-          else
-            @log << "#{company.owner.name} cannot afford #{format_currency(payment)} for #{company.name}"
-            close_p16
+          if associated_minor?(corporation)
+            major = associated_major(corporation)
+            hex_by_id(corporation.coordinates).tile.cities[0].add_reservation!(major)
+            @log << "#{major.name} reservation takes the place of #{corporation.name}"
+          elsif regional_railway?(corporation)
+            company = company_by_id("M#{corporation.id}")
```

Comment:
helper method for the magic

Reply:
Made method "company_id_from_corp_id" for this